kore/src/Kore/Strategies/Goal.hs

```diff
@@ -1008,6 +988,15 @@ removalPredicate
           , Pretty.indent 4 (unparse destTerm)
           ]
   where
+    evaluateMultipleExists patt = do
+        let extraElemVariables =
```

Comment:
I don't think this is quantifying over the correct variables. I think we have an explicit list of variables to quantify in the claim.